Implement and Apply TZone Allocation Macros

msaboff opened the pull request:
#### a9681223364eec9a18680e158395ed3bbdd7952e
<pre>
Implement and Apply TZone Allocation Macros
<a href="https://rdar.apple.com/114836775">rdar://114836775</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=266095">https://bugs.webkit.org/show_bug.cgi?id=266095</a>

Reviewed by Mark Lam.

Added initial implementation of sized typed allocators, aka TZone.
This change adds a basic sized type allocation system and the macros used so a class will use a TZone allocation.
It also changes fixed sized JavaScript classes to use those macros.

For Classes that don&apos;t have a corresponding .cpp file, a per-functional group XXXTZoneImpls.cpp file is created in
the function directory for the implementations of such TZone allocated classes.

The new Platform USE directive USE_TZONE_MALLOC enables building with TZone allocation or FastMalloc allocation.
This change defaults to using FastMalloc allocaiton.  Non-ARM64 builds have TZone allocations disabled as they
haven&apos;t been changed and built yet.

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/assembler/AbstractMacroAssembler.h:
* Source/JavaScriptCore/assembler/AssemblyComments.cpp:
* Source/JavaScriptCore/assembler/AssemblyComments.h:
* Source/JavaScriptCore/assembler/LinkBuffer.cpp:
* Source/JavaScriptCore/assembler/LinkBuffer.h:
* Source/JavaScriptCore/assembler/MacroAssemblerARM64.cpp:
* Source/JavaScriptCore/assembler/MacroAssemblerARM64.h:
* Source/JavaScriptCore/assembler/PerfLog.cpp:
* Source/JavaScriptCore/assembler/PerfLog.h:
* Source/JavaScriptCore/assembler/ProbeContext.cpp:
* Source/JavaScriptCore/assembler/ProbeContext.h:
* Source/JavaScriptCore/assembler/ProbeStack.cpp:
* Source/JavaScriptCore/assembler/ProbeStack.h:
* Source/JavaScriptCore/b3/B3BackwardsCFG.h:
* Source/JavaScriptCore/b3/B3BackwardsDominators.h:
* Source/JavaScriptCore/b3/B3BasicBlock.cpp:
* Source/JavaScriptCore/b3/B3BasicBlock.h:
* Source/JavaScriptCore/b3/B3CFG.h:
* Source/JavaScriptCore/b3/B3CheckSpecial.cpp:
* Source/JavaScriptCore/b3/B3CheckSpecial.h:
* Source/JavaScriptCore/b3/B3DataSection.cpp:
* Source/JavaScriptCore/b3/B3DataSection.h:
* Source/JavaScriptCore/b3/B3Dominators.h:
* Source/JavaScriptCore/b3/B3NaturalLoops.h:
* Source/JavaScriptCore/b3/B3PhiChildren.cpp:
* Source/JavaScriptCore/b3/B3PhiChildren.h:
* Source/JavaScriptCore/b3/B3Procedure.cpp:
* Source/JavaScriptCore/b3/B3Procedure.h:
* Source/JavaScriptCore/b3/B3StackmapSpecial.cpp:
* Source/JavaScriptCore/b3/B3StackmapSpecial.h:
* Source/JavaScriptCore/b3/B3TZoneImpls.cpp: Copied from Source/JavaScriptCore/b3/B3BackwardsDominators.h.
* Source/JavaScriptCore/b3/B3Value.cpp:
* Source/JavaScriptCore/b3/B3Value.h:
* Source/JavaScriptCore/b3/B3ValueRep.cpp:
* Source/JavaScriptCore/b3/B3ValueRep.h:
* Source/JavaScriptCore/b3/B3Variable.cpp:
* Source/JavaScriptCore/b3/B3Variable.h:
* Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackAndGenerateCode.cpp:
* Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackAndGenerateCode.h:
* Source/JavaScriptCore/b3/air/AirBasicBlock.cpp:
* Source/JavaScriptCore/b3/air/AirBasicBlock.h:
* Source/JavaScriptCore/b3/air/AirCCallSpecial.cpp:
* Source/JavaScriptCore/b3/air/AirCCallSpecial.h:
* Source/JavaScriptCore/b3/air/AirCFG.h:
* Source/JavaScriptCore/b3/air/AirCode.cpp:
* Source/JavaScriptCore/b3/air/AirCode.h:
* Source/JavaScriptCore/b3/air/AirDisassembler.cpp:
* Source/JavaScriptCore/b3/air/AirDisassembler.h:
* Source/JavaScriptCore/b3/air/AirLiveness.h:
* Source/JavaScriptCore/b3/air/AirLivenessAdapter.h:
* Source/JavaScriptCore/b3/air/AirPrintSpecial.cpp:
* Source/JavaScriptCore/b3/air/AirPrintSpecial.h:
* Source/JavaScriptCore/b3/air/AirSpecial.cpp:
* Source/JavaScriptCore/b3/air/AirSpecial.h:
* Source/JavaScriptCore/b3/air/AirStackSlot.cpp:
* Source/JavaScriptCore/b3/air/AirStackSlot.h:
* Source/JavaScriptCore/b3/air/AirTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/JavaScriptCore/builtins/BuiltinExecutables.cpp:
* Source/JavaScriptCore/builtins/BuiltinExecutables.h:
* Source/JavaScriptCore/builtins/BuiltinNames.cpp:
* Source/JavaScriptCore/builtins/BuiltinNames.h:
* Source/JavaScriptCore/bytecode/AccessCaseSnippetParams.cpp:
* Source/JavaScriptCore/bytecode/AccessCaseSnippetParams.h:
* Source/JavaScriptCore/bytecode/AdaptiveInferredPropertyValueWatchpointBase.cpp:
* Source/JavaScriptCore/bytecode/AdaptiveInferredPropertyValueWatchpointBase.h:
* Source/JavaScriptCore/bytecode/BytecodeBasicBlock.h:
* Source/JavaScriptCore/bytecode/BytecodeGraph.h:
* Source/JavaScriptCore/bytecode/BytecodeIntrinsicRegistry.cpp:
* Source/JavaScriptCore/bytecode/BytecodeIntrinsicRegistry.h:
* Source/JavaScriptCore/bytecode/BytecodeLivenessAnalysis.cpp:
* Source/JavaScriptCore/bytecode/BytecodeLivenessAnalysis.h:
* Source/JavaScriptCore/bytecode/CallLinkStatus.cpp:
* Source/JavaScriptCore/bytecode/CallLinkStatus.h:
* Source/JavaScriptCore/bytecode/CallVariant.h:
* Source/JavaScriptCore/bytecode/CheckPrivateBrandStatus.cpp:
* Source/JavaScriptCore/bytecode/CheckPrivateBrandStatus.h:
* Source/JavaScriptCore/bytecode/CheckPrivateBrandVariant.h:
* Source/JavaScriptCore/bytecode/CodeOrigin.cpp:
* Source/JavaScriptCore/bytecode/CodeOrigin.h:
* Source/JavaScriptCore/bytecode/DeferredSourceDump.h:
* Source/JavaScriptCore/bytecode/DeleteByStatus.cpp:
* Source/JavaScriptCore/bytecode/DeleteByStatus.h:
* Source/JavaScriptCore/bytecode/DeleteByVariant.h:
* Source/JavaScriptCore/bytecode/ExecutionCounter.cpp:
* Source/JavaScriptCore/bytecode/ExecutionCounter.h:
* Source/JavaScriptCore/bytecode/FullBytecodeLiveness.h:
* Source/JavaScriptCore/bytecode/GetByVariant.h:
* Source/JavaScriptCore/bytecode/InByStatus.cpp:
* Source/JavaScriptCore/bytecode/InByStatus.h:
* Source/JavaScriptCore/bytecode/InByVariant.h:
* Source/JavaScriptCore/bytecode/InstanceOfStatus.h:
* Source/JavaScriptCore/bytecode/InstanceOfVariant.h:
* Source/JavaScriptCore/bytecode/MetadataTable.h:
* Source/JavaScriptCore/bytecode/PutByVariant.h:
* Source/JavaScriptCore/bytecode/SetPrivateBrandStatus.cpp:
* Source/JavaScriptCore/bytecode/SetPrivateBrandStatus.h:
* Source/JavaScriptCore/bytecode/SetPrivateBrandVariant.h:
* Source/JavaScriptCore/bytecode/StructureStubClearingWatchpoint.cpp:
* Source/JavaScriptCore/bytecode/StructureStubClearingWatchpoint.h:
* Source/JavaScriptCore/bytecode/StructureStubInfo.h:
* Source/JavaScriptCore/bytecode/UnlinkedCodeBlockGenerator.cpp:
* Source/JavaScriptCore/bytecode/UnlinkedCodeBlockGenerator.h:
* Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp:
* Source/JavaScriptCore/bytecompiler/BytecodeGenerator.h:
* Source/JavaScriptCore/config.h:
* Source/JavaScriptCore/debugger/Breakpoint.h:
* Source/JavaScriptCore/debugger/Debugger.cpp:
* Source/JavaScriptCore/debugger/Debugger.h:
* Source/JavaScriptCore/dfg/DFGAbstractInterpreter.h:
* Source/JavaScriptCore/dfg/DFGAdaptiveInferredPropertyValueWatchpoint.cpp:
* Source/JavaScriptCore/dfg/DFGAdaptiveInferredPropertyValueWatchpoint.h:
* Source/JavaScriptCore/dfg/DFGArrayifySlowPathGenerator.h:
* Source/JavaScriptCore/dfg/DFGBackwardsCFG.h:
* Source/JavaScriptCore/dfg/DFGBackwardsDominators.h:
* Source/JavaScriptCore/dfg/DFGBasicBlock.cpp:
* Source/JavaScriptCore/dfg/DFGBasicBlock.h:
* Source/JavaScriptCore/dfg/DFGBlockMap.h:
* Source/JavaScriptCore/dfg/DFGCFG.h:
* Source/JavaScriptCore/dfg/DFGCSEPhase.cpp:
* Source/JavaScriptCore/dfg/DFGCallArrayAllocatorSlowPathGenerator.h:
* Source/JavaScriptCore/dfg/DFGCallCreateDirectArgumentsSlowPathGenerator.h:
* Source/JavaScriptCore/dfg/DFGCombinedLiveness.cpp:
* Source/JavaScriptCore/dfg/DFGCombinedLiveness.h:
* Source/JavaScriptCore/dfg/DFGControlEquivalenceAnalysis.h:
* Source/JavaScriptCore/dfg/DFGDisassembler.cpp:
* Source/JavaScriptCore/dfg/DFGDisassembler.h:
* Source/JavaScriptCore/dfg/DFGDominators.h:
* Source/JavaScriptCore/dfg/DFGFailedFinalizer.cpp:
* Source/JavaScriptCore/dfg/DFGFailedFinalizer.h:
* Source/JavaScriptCore/dfg/DFGFinalizer.cpp:
* Source/JavaScriptCore/dfg/DFGFinalizer.h:
* Source/JavaScriptCore/dfg/DFGFlowIndexing.cpp:
* Source/JavaScriptCore/dfg/DFGFlowIndexing.h:
* Source/JavaScriptCore/dfg/DFGFlowMap.h:
* Source/JavaScriptCore/dfg/DFGInPlaceAbstractState.cpp:
* Source/JavaScriptCore/dfg/DFGInPlaceAbstractState.h:
* Source/JavaScriptCore/dfg/DFGJITCompiler.cpp:
* Source/JavaScriptCore/dfg/DFGJITCompiler.h:
* Source/JavaScriptCore/dfg/DFGJITFinalizer.cpp:
* Source/JavaScriptCore/dfg/DFGJITFinalizer.h:
* Source/JavaScriptCore/dfg/DFGNaturalLoops.h:
* Source/JavaScriptCore/dfg/DFGNullAbstractState.h:
* Source/JavaScriptCore/dfg/DFGPhiChildren.cpp:
* Source/JavaScriptCore/dfg/DFGPhiChildren.h:
* Source/JavaScriptCore/dfg/DFGSaneStringGetByValSlowPathGenerator.h:
* Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp:
* Source/JavaScriptCore/dfg/DFGSpeculativeJIT.h:
* Source/JavaScriptCore/dfg/DFGTZoneImpls.cpp: Copied from Source/JavaScriptCore/dfg/DFGFlowIndexing.cpp.
* Source/JavaScriptCore/dfg/DFGVariableAccessData.h:
* Source/JavaScriptCore/disassembler/ARM64/A64DOpcode.cpp:
* Source/JavaScriptCore/disassembler/ARM64/A64DOpcode.h:
* Source/JavaScriptCore/disassembler/Disassembler.cpp:
* Source/JavaScriptCore/ftl/FTLAbstractHeap.cpp:
* Source/JavaScriptCore/ftl/FTLAbstractHeap.h:
* Source/JavaScriptCore/ftl/FTLJITFinalizer.cpp:
* Source/JavaScriptCore/ftl/FTLJITFinalizer.h:
* Source/JavaScriptCore/ftl/FTLLazySlowPath.cpp:
* Source/JavaScriptCore/ftl/FTLLazySlowPath.h:
* Source/JavaScriptCore/ftl/FTLThunks.cpp:
* Source/JavaScriptCore/ftl/FTLThunks.h:
* Source/JavaScriptCore/heap/AbstractSlotVisitor.h:
* Source/JavaScriptCore/heap/AlignedMemoryAllocator.cpp:
* Source/JavaScriptCore/heap/BlockDirectoryBits.h:
* Source/JavaScriptCore/heap/CodeBlockSet.cpp:
* Source/JavaScriptCore/heap/CodeBlockSet.h:
* Source/JavaScriptCore/heap/GCSegmentedArray.cpp:
* Source/JavaScriptCore/heap/GCSegmentedArray.h:
* Source/JavaScriptCore/heap/HeapCellType.h:
* Source/JavaScriptCore/heap/HeapProfiler.cpp:
* Source/JavaScriptCore/heap/HeapProfiler.h:
* Source/JavaScriptCore/heap/HeapSnapshot.cpp:
* Source/JavaScriptCore/heap/HeapSnapshot.h:
* Source/JavaScriptCore/heap/HeapSnapshotBuilder.cpp:
* Source/JavaScriptCore/heap/HeapSnapshotBuilder.h:
* Source/JavaScriptCore/heap/IsoHeapCellType.cpp:
* Source/JavaScriptCore/heap/IsoHeapCellType.h:
* Source/JavaScriptCore/heap/IsoSubspace.cpp:
* Source/JavaScriptCore/heap/IsoSubspace.h:
* Source/JavaScriptCore/heap/JITStubRoutineSet.cpp:
* Source/JavaScriptCore/heap/JITStubRoutineSet.h:
* Source/JavaScriptCore/heap/MachineStackMarker.cpp:
* Source/JavaScriptCore/heap/MachineStackMarker.h:
* Source/JavaScriptCore/heap/MarkStackMergingConstraint.cpp:
* Source/JavaScriptCore/heap/MarkStackMergingConstraint.h:
* Source/JavaScriptCore/heap/MarkingConstraint.cpp:
* Source/JavaScriptCore/heap/MarkingConstraint.h:
* Source/JavaScriptCore/heap/MarkingConstraintSet.cpp:
* Source/JavaScriptCore/heap/MarkingConstraintSet.h:
* Source/JavaScriptCore/heap/MarkingConstraintSolver.h:
* Source/JavaScriptCore/heap/MutatorScheduler.cpp:
* Source/JavaScriptCore/heap/MutatorScheduler.h:
* Source/JavaScriptCore/heap/SimpleMarkingConstraint.cpp:
* Source/JavaScriptCore/heap/SimpleMarkingConstraint.h:
* Source/JavaScriptCore/heap/SlotVisitor.cpp:
* Source/JavaScriptCore/heap/SlotVisitor.h:
* Source/JavaScriptCore/heap/SpaceTimeMutatorScheduler.cpp:
* Source/JavaScriptCore/heap/SpaceTimeMutatorScheduler.h:
* Source/JavaScriptCore/heap/StochasticSpaceTimeMutatorScheduler.cpp:
* Source/JavaScriptCore/heap/StochasticSpaceTimeMutatorScheduler.h:
* Source/JavaScriptCore/heap/Subspace.cpp:
* Source/JavaScriptCore/heap/Subspace.h:
* Source/JavaScriptCore/heap/SynchronousStopTheWorldMutatorScheduler.cpp:
* Source/JavaScriptCore/heap/SynchronousStopTheWorldMutatorScheduler.h:
* Source/JavaScriptCore/heap/VerifierSlotVisitor.cpp:
* Source/JavaScriptCore/heap/VerifierSlotVisitor.h:
* Source/JavaScriptCore/inspector/ConsoleMessage.cpp:
* Source/JavaScriptCore/inspector/ConsoleMessage.h:
* Source/JavaScriptCore/inspector/InjectedScriptManager.cpp:
* Source/JavaScriptCore/inspector/InjectedScriptManager.h:
* Source/JavaScriptCore/inspector/InspectorAgentBase.h:
* Source/JavaScriptCore/inspector/JSGlobalObjectConsoleClient.cpp:
* Source/JavaScriptCore/inspector/JSGlobalObjectConsoleClient.h:
* Source/JavaScriptCore/inspector/JSGlobalObjectDebugger.cpp:
* Source/JavaScriptCore/inspector/JSGlobalObjectDebugger.h:
* Source/JavaScriptCore/inspector/JSGlobalObjectInspectorController.cpp:
* Source/JavaScriptCore/inspector/JSGlobalObjectInspectorController.h:
* Source/JavaScriptCore/inspector/JSInjectedScriptHost.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorAuditAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorAuditAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorConsoleAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorConsoleAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorHeapAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorHeapAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorRuntimeAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorRuntimeAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorScriptProfilerAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorScriptProfilerAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorTargetAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorTargetAgent.h:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectAuditAgent.cpp:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectAuditAgent.h:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectDebuggerAgent.cpp:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectDebuggerAgent.h:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectRuntimeAgent.cpp:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectRuntimeAgent.h:
* Source/JavaScriptCore/interpreter/CheckpointOSRExitSideState.h:
* Source/JavaScriptCore/interpreter/Interpreter.h:
* Source/JavaScriptCore/interpreter/Register.h:
* Source/JavaScriptCore/interpreter/ShadowChicken.cpp:
* Source/JavaScriptCore/interpreter/ShadowChicken.h:
* Source/JavaScriptCore/jit/AssemblyHelpers.cpp:
* Source/JavaScriptCore/jit/AssemblyHelpers.h:
* Source/JavaScriptCore/jit/CCallHelpers.cpp:
* Source/JavaScriptCore/jit/CCallHelpers.h:
* Source/JavaScriptCore/jit/CallFrameShuffleData.cpp:
* Source/JavaScriptCore/jit/CallFrameShuffleData.h:
* Source/JavaScriptCore/jit/CallFrameShuffler.h:
* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
* Source/JavaScriptCore/jit/ExecutableAllocator.h:
* Source/JavaScriptCore/jit/ICStats.cpp:
* Source/JavaScriptCore/jit/ICStats.h:
* Source/JavaScriptCore/jit/JIT.cpp:
* Source/JavaScriptCore/jit/JIT.h:
* Source/JavaScriptCore/jit/JITCompilation.cpp:
* Source/JavaScriptCore/jit/JITCompilation.h:
* Source/JavaScriptCore/jit/JITDisassembler.cpp:
* Source/JavaScriptCore/jit/JITDisassembler.h:
* Source/JavaScriptCore/jit/JITMathIC.h:
* Source/JavaScriptCore/jit/JITOpaqueByproduct.h:
* Source/JavaScriptCore/jit/JITOpaqueByproducts.cpp:
* Source/JavaScriptCore/jit/JITOpaqueByproducts.h:
* Source/JavaScriptCore/jit/JITSizeStatistics.cpp:
* Source/JavaScriptCore/jit/JITSizeStatistics.h:
* Source/JavaScriptCore/jit/JITTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/JavaScriptCore/jit/JITThunks.cpp:
* Source/JavaScriptCore/jit/JITThunks.h:
* Source/JavaScriptCore/jit/JITWorklist.cpp:
* Source/JavaScriptCore/jit/JITWorklist.h:
* Source/JavaScriptCore/jit/JSInterfaceJIT.h:
* Source/JavaScriptCore/jit/PCToCodeOriginMap.cpp:
* Source/JavaScriptCore/jit/PCToCodeOriginMap.h:
* Source/JavaScriptCore/jit/SpecializedThunkJIT.h:
* Source/JavaScriptCore/jsc.cpp:
* Source/JavaScriptCore/llint/LLIntOffsetsExtractor.cpp:
* Source/JavaScriptCore/parser/Lexer.cpp:
* Source/JavaScriptCore/parser/Lexer.h:
* Source/JavaScriptCore/parser/ModuleScopeData.h:
* Source/JavaScriptCore/parser/Parser.cpp:
* Source/JavaScriptCore/parser/Parser.h:
* Source/JavaScriptCore/parser/VariableEnvironment.cpp:
* Source/JavaScriptCore/parser/VariableEnvironment.h:
* Source/JavaScriptCore/profiler/ProfilerDatabase.cpp:
* Source/JavaScriptCore/profiler/ProfilerDatabase.h:
* Source/JavaScriptCore/profiler/ProfilerExecutionCounter.h:
* Source/JavaScriptCore/profiler/ProfilerTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/JavaScriptCore/runtime/ArgList.cpp:
* Source/JavaScriptCore/runtime/ArgList.h:
* Source/JavaScriptCore/runtime/ArrayBuffer.h:
* Source/JavaScriptCore/runtime/BasicBlockLocation.cpp:
* Source/JavaScriptCore/runtime/BasicBlockLocation.h:
* Source/JavaScriptCore/runtime/BufferMemoryHandle.cpp:
* Source/JavaScriptCore/runtime/BufferMemoryHandle.h:
* Source/JavaScriptCore/runtime/CodeCache.cpp:
* Source/JavaScriptCore/runtime/CodeCache.h:
* Source/JavaScriptCore/runtime/CommonIdentifiers.cpp:
* Source/JavaScriptCore/runtime/CommonIdentifiers.h:
* Source/JavaScriptCore/runtime/ControlFlowProfiler.cpp:
* Source/JavaScriptCore/runtime/ControlFlowProfiler.h:
* Source/JavaScriptCore/runtime/DOMAnnotation.h:
* Source/JavaScriptCore/runtime/DeferredWorkTimer.cpp:
* Source/JavaScriptCore/runtime/DeferredWorkTimer.h:
* Source/JavaScriptCore/runtime/DoublePredictionFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/DoublePredictionFuzzerAgent.h:
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgent.h:
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgentBase.h:
* Source/JavaScriptCore/runtime/GenericOffset.h:
* Source/JavaScriptCore/runtime/ISO8601.h:
* Source/JavaScriptCore/runtime/IntlCache.cpp:
* Source/JavaScriptCore/runtime/IntlCache.h:
* Source/JavaScriptCore/runtime/IntlNumberFormat.h:
* Source/JavaScriptCore/runtime/JSDateMath.cpp:
* Source/JavaScriptCore/runtime/JSDateMath.h:
* Source/JavaScriptCore/runtime/JSDestructibleObjectHeapCellType.cpp:
* Source/JavaScriptCore/runtime/JSDestructibleObjectHeapCellType.h:
* Source/JavaScriptCore/runtime/JSGlobalObject.h:
(JSC::JSGlobalObject::customGetterFunctionSet):
(JSC::JSGlobalObject::customSetterFunctionSet):
* Source/JavaScriptCore/runtime/JSGlobalObjectDebuggable.cpp:
* Source/JavaScriptCore/runtime/JSGlobalObjectDebuggable.h:
* Source/JavaScriptCore/runtime/JSRunLoopTimer.cpp:
* Source/JavaScriptCore/runtime/JSRunLoopTimer.h:
* Source/JavaScriptCore/runtime/MegamorphicCache.cpp:
* Source/JavaScriptCore/runtime/MegamorphicCache.h:
* Source/JavaScriptCore/runtime/NarrowingNumberPredictionFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/NarrowingNumberPredictionFuzzerAgent.h:
* Source/JavaScriptCore/runtime/NativeCallee.h:
* Source/JavaScriptCore/runtime/NativeCalleeRegistry.h:
* Source/JavaScriptCore/runtime/ObjectPropertyChangeAdaptiveWatchpoint.h:
* Source/JavaScriptCore/runtime/Options.cpp:
* Source/JavaScriptCore/runtime/PredictionFileCreatingFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/PredictionFileCreatingFuzzerAgent.h:
* Source/JavaScriptCore/runtime/PropertyDescriptor.cpp:
* Source/JavaScriptCore/runtime/RandomizingFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/RandomizingFuzzerAgent.h:
* Source/JavaScriptCore/runtime/RegExpCache.cpp:
* Source/JavaScriptCore/runtime/RegExpCache.h:
* Source/JavaScriptCore/runtime/RuntimeTZoneImpls.cpp: Copied from Source/JavaScriptCore/heap/IsoHeapCellType.cpp.
* Source/JavaScriptCore/runtime/SamplingProfiler.cpp:
* Source/JavaScriptCore/runtime/SamplingProfiler.h:
* Source/JavaScriptCore/runtime/SparseArrayValueMap.h:
* Source/JavaScriptCore/runtime/StringReplaceCache.h:
* Source/JavaScriptCore/runtime/StringSplitCache.h:
* Source/JavaScriptCore/runtime/StructureRareData.cpp:
* Source/JavaScriptCore/runtime/TypeProfiler.cpp:
* Source/JavaScriptCore/runtime/TypeProfiler.h:
* Source/JavaScriptCore/runtime/TypeProfilerLog.cpp:
* Source/JavaScriptCore/runtime/TypeProfilerLog.h:
* Source/JavaScriptCore/runtime/VM.h:
* Source/JavaScriptCore/runtime/WaiterListManager.cpp:
* Source/JavaScriptCore/runtime/WaiterListManager.h:
* Source/JavaScriptCore/runtime/Watchdog.cpp:
* Source/JavaScriptCore/runtime/Watchdog.h:
* Source/JavaScriptCore/runtime/WeakGCSet.h:
* Source/JavaScriptCore/runtime/WideningNumberPredictionFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/WideningNumberPredictionFuzzerAgent.h:
* Source/JavaScriptCore/runtime/WriteBarrier.h:
* Source/JavaScriptCore/tools/CellList.h:
* Source/JavaScriptCore/tools/CompilerTimingScope.cpp:
* Source/JavaScriptCore/tools/HeapVerifier.cpp:
* Source/JavaScriptCore/tools/HeapVerifier.h:
* Source/JavaScriptCore/tools/JSDollarVM.cpp:
* Source/JavaScriptCore/tools/VMInspector.cpp:
* Source/JavaScriptCore/tools/VMInspector.h:
* Source/JavaScriptCore/wasm/WasmB3IRGenerator.cpp:
* Source/JavaScriptCore/wasm/WasmBBQDisassembler.cpp:
* Source/JavaScriptCore/wasm/WasmBBQDisassembler.h:
* Source/JavaScriptCore/wasm/WasmCallee.cpp:
* Source/JavaScriptCore/wasm/WasmCallee.h:
* Source/JavaScriptCore/wasm/WasmFormat.h:
* Source/JavaScriptCore/wasm/WasmFunctionCodeBlockGenerator.cpp:
* Source/JavaScriptCore/wasm/WasmFunctionCodeBlockGenerator.h:
* Source/JavaScriptCore/wasm/WasmFunctionIPIntMetadataGenerator.cpp:
* Source/JavaScriptCore/wasm/WasmFunctionIPIntMetadataGenerator.h:
* Source/JavaScriptCore/wasm/WasmFunctionParser.h:
* Source/JavaScriptCore/wasm/WasmGlobal.cpp:
* Source/JavaScriptCore/wasm/WasmGlobal.h:
* Source/JavaScriptCore/wasm/WasmMemory.cpp:
(): Deleted.
* Source/JavaScriptCore/wasm/WasmMemory.h:
* Source/JavaScriptCore/wasm/WasmOSREntryData.h:
* Source/JavaScriptCore/wasm/WasmStreamingParser.h:
* Source/JavaScriptCore/wasm/WasmTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/JavaScriptCore/wasm/WasmTable.cpp:
* Source/JavaScriptCore/wasm/WasmTable.h:
* Source/JavaScriptCore/wasm/WasmTag.cpp:
* Source/JavaScriptCore/wasm/WasmTag.h:
* Source/JavaScriptCore/wasm/WasmThunks.cpp:
* Source/JavaScriptCore/wasm/WasmThunks.h:
* Source/JavaScriptCore/wasm/WasmTierUpCount.cpp:
* Source/JavaScriptCore/wasm/WasmTierUpCount.h:
* Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp:
(JSC::Wasm::TypeDefinition::tryCreateFunctionSignature):
(JSC::Wasm::TypeDefinition::tryCreateStructType):
(JSC::Wasm::TypeDefinition::tryCreateArrayType):
(JSC::Wasm::TypeDefinition::tryCreateRecursionGroup):
(JSC::Wasm::TypeDefinition::tryCreateProjection):
(JSC::Wasm::TypeDefinition::tryCreateSubtype):
* Source/JavaScriptCore/wasm/WasmTypeDefinition.h:
* Source/JavaScriptCore/wasm/WasmValueLocation.h:
* Source/JavaScriptCore/wasm/WasmWorklist.cpp:
* Source/JavaScriptCore/wasm/WasmWorklist.h:
* Source/JavaScriptCore/wasm/js/WebAssemblyFunction.cpp:
* Source/JavaScriptCore/yarr/RegularExpression.cpp:
* Source/JavaScriptCore/yarr/RegularExpression.h:
* Source/JavaScriptCore/yarr/YarrDisassembler.cpp:
* Source/JavaScriptCore/yarr/YarrDisassembler.h:
* Source/JavaScriptCore/yarr/YarrInterpreter.cpp:
* Source/JavaScriptCore/yarr/YarrInterpreter.h:
* Source/JavaScriptCore/yarr/YarrJIT.cpp:
* Source/JavaScriptCore/yarr/YarrJIT.h:
* Source/JavaScriptCore/yarr/YarrPattern.h:
* Source/JavaScriptCore/yarr/YarrTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/CMakeLists.txt:
* Source/WTF/wtf/PlatformUse.h:
* Source/WTF/wtf/TZoneMalloc.h: Copied from Source/JavaScriptCore/jit/JITSizeStatistics.h.
* Source/WTF/wtf/TZoneMallocInlines.h: Copied from Source/JavaScriptCore/runtime/StringSplitCache.h.
* Source/bmalloc/bmalloc.xcodeproj/project.pbxproj:
* Source/bmalloc/bmalloc/TZoneHeap.cpp: Copied from Source/JavaScriptCore/runtime/StringReplaceCache.h.
(bmalloc::api::tzoneAllocate):
(bmalloc::api::tzoneTryAllocate):
(bmalloc::api::tzoneDeallocate):
* Source/bmalloc/bmalloc/TZoneHeap.h: Added.
(bmalloc::api::TZoneHeap::TZoneHeap):
(bmalloc::api::TZoneHeap::allocate):
(bmalloc::api::TZoneHeap::tryAllocate):
(bmalloc::api::TZoneHeap::deallocate):
(bmalloc::api::TZoneHeap::scavenge):
(bmalloc::api::TZoneHeap::initialize):
(bmalloc::api::TZoneHeap::isInitialized):
(bmalloc::api::TZoneHeap::provideHeap):
(bmalloc::api::TZoneHeap::allocatorOffset):
(bmalloc::api::TZoneHeap::setAllocatorOffset):
(bmalloc::api::TZoneHeap::deallocatorOffset):
(bmalloc::api::TZoneHeap::setDeallocatorOffset):
* Source/bmalloc/bmalloc/TZoneHeapInlines.h: Added.
(bmalloc::api::IsoHeap&lt;Type&gt;::TZoneHeap):
(bmalloc::api::TZoneHeap&lt;Type&gt;::allocate):
(bmalloc::api::TZoneHeap&lt;Type&gt;::tryAllocate):
(bmalloc::api::TZoneHeap&lt;Type&gt;::deallocate):
(bmalloc::api::TZoneHeap&lt;Type&gt;::scavenge):
(bmalloc::api::TZoneHeap&lt;Type&gt;::isInitialized):
(bmalloc::api::TZoneHeap&lt;Type&gt;::initialize):
(bmalloc::api::TZoneHeap&lt;Type&gt;::impl):

Canonical link: <a href="https://commits.webkit.org/271904@main">https://commits.webkit.org/271904@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ca36dbd05d14b60b94aa02948dea9c6d54401965

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29973 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8639 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31291 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32483 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27102 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30633 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10828 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5893 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27148 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30267 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7246 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25576 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6182 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6345 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26665 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33819 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/25748 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27341 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27097 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32525 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/30142 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6286 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4457 "Found 1 new test failure: imported/w3c/web-platform-tests/video-rvfc/request-video-frame-callback-webrtc.https.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30321 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8024 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/36539 "Built successfully") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/7109 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7029 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7883 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3874 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6813 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
```diff
@@ -4374,6 +4374,13 @@
 		6560A4CF04B3B3E7008AE952 /* CoreFoundation.framework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.framework; name = CoreFoundation.framework; path = /System/Library/Frameworks/CoreFoundation.framework; sourceTree = "<absolute>"; };
 		65621E6B089E859700760F35 /* PropertySlot.cpp */ = {isa = PBXFileReference; fileEncoding = 30; indentWidth = 4; lastKnownFileType = sourcecode.cpp.cpp; path = PropertySlot.cpp; sourceTree = "<group>"; tabWidth = 8; };
 		65621E6C089E859700760F35 /* PropertySlot.h */ = {isa = PBXFileReference; fileEncoding = 30; indentWidth = 4; lastKnownFileType = sourcecode.c.h; path = PropertySlot.h; sourceTree = "<group>"; tabWidth = 8; };
+		6571E94C2AF2E769009DF224 /* RuntimeTZoneImpls.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = RuntimeTZoneImpls.cpp; sourceTree = "<group>"; };
```
Typo in commit msg: "allocaiton" -> "allocation"

MenloDorian commented on Source/JavaScriptCore/assembler/AbstractMacroAssembler.cpp:
```diff
@@ -30,9 +30,12 @@
 
 namespace JSC {
 
+WTF_MAKE_TZONE_ALLOCATED_IMPL(AbstractMacroAssemblerBase);
```
I suspect we can omit this since we never instantiate this class.  We want the declaration in the .h though.  If anyone tries to instantiate this, they'll just get a linker error.  That said, out linker will probably elide this if it's not used.  So, maybe it's better to be uniform with adding the IMPLs after all.

msaboff replied:
Fixed.

MenloDorian commented on Source/JavaScriptCore/assembler/ProbeStack.h:
```diff
@@ -37,7 +38,7 @@ namespace JSC {
 
 class Page {
-    WTF_MAKE_FAST_ALLOCATED;
+    WTF_MAKE_TZONE_ALLOCATED(Page); // mlam Make not heap allocated.
```
Remove all the `// mlam` comments please.  They were placed there just as a reminder of needed action.

msaboff replied:
Fixed.

MenloDorian commented on Source/JavaScriptCore/parser/Lexer.cpp:
```diff
@@ -35,10 +35,13 @@
 namespace JSC {
 
+
+
```
Remove these empty lines.

msaboff replied:
Fixed.

MenloDorian commented on Source/JavaScriptCore/runtime/JSRunLoopTimer.h:
```diff
@@ -65,7 +66,8 @@ class JSRunLoopTimer : public ThreadSafeRefCounted<JSRunLoopTimer> {
         class PerVMData {
-            WTF_MAKE_FAST_ALLOCATED;
+            WTF_MAKE_FAST_ALLOCATED; // mlam FIXME
+//            WTF_MAKE_TZONE_ALLOCATED(PerVMData);
```
Please remove the FIXME and commented out line of code.

You should just grep for "mlam" and remove all such comments.

msaboff replied:
Fixed.

MenloDorian commented on Source/JavaScriptCore/runtime/StructureCache.cpp:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2013-2019 Apple Inc. All rights reserved.
+ * Copyright (C) 2013-2023 Apple Inc. All rights reserved.
```
Looks like there's no other change in this file.  So, this is not needed.

msaboff replied:
Fixed.

MenloDorian commented on Source/JavaScriptCore/runtime/WeakGCMap.h:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2009, 2015-2016 Apple Inc. All rights reserved.
+ * Copyright (C) 2009-2023 Apple Inc. All rights reserved.
```
Not needed.

msaboff replied:
Fixed.

MenloDorian commented on Source/JavaScriptCore/unexported-libc++.txt:
```diff
@@ -27,4 +27,4 @@ __ZT[IS]St*
 
 # global operator delete and delete[]
-__Zd[la]*
+#  &&&&  __Zd[la]*
```
Is this really needed?

msaboff replied:
Fixed.

MenloDorian reviewed (approved):
r=me with fixes, and if EWS bots are green.

msaboff committed:
Implement and Apply TZone Allocation Macros
rdar://114836775
https://bugs.webkit.org/show_bug.cgi?id=266095

Reviewed by Mark Lam.

Added initial implementation of sized typed allocators, aka TZone.
This change adds a basic sized type allocation system and the macros used so a class will use a TZone allocation.
It also changes fixed sized JavaScript classes to use those macros.

For Classes that don't have a corresponding .cpp file, a per-functional group XXXTZoneImpls.cpp file is created in
the function directory for the implementations of such TZone allocated classes.

The new Platform USE directive USE_TZONE_MALLOC enables building with TZone allocation or FastMalloc allocation.
This change defaults to using FastMalloc allocaiton.  Non-ARM64 builds have TZone allocations disabled as they
haven't been changed and built yet.

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/assembler/AbstractMacroAssembler.h:
* Source/JavaScriptCore/assembler/AssemblyComments.cpp:
* Source/JavaScriptCore/assembler/AssemblyComments.h:
* Source/JavaScriptCore/assembler/LinkBuffer.cpp:
* Source/JavaScriptCore/assembler/LinkBuffer.h:
* Source/JavaScriptCore/assembler/MacroAssemblerARM64.cpp:
* Source/JavaScriptCore/assembler/MacroAssemblerARM64.h:
* Source/JavaScriptCore/assembler/PerfLog.cpp:
* Source/JavaScriptCore/assembler/PerfLog.h:
* Source/JavaScriptCore/assembler/ProbeContext.cpp:
* Source/JavaScriptCore/assembler/ProbeContext.h:
* Source/JavaScriptCore/assembler/ProbeStack.cpp:
* Source/JavaScriptCore/assembler/ProbeStack.h:
* Source/JavaScriptCore/b3/B3BackwardsCFG.h:
* Source/JavaScriptCore/b3/B3BackwardsDominators.h:
* Source/JavaScriptCore/b3/B3BasicBlock.cpp:
* Source/JavaScriptCore/b3/B3BasicBlock.h:
* Source/JavaScriptCore/b3/B3CFG.h:
* Source/JavaScriptCore/b3/B3CheckSpecial.cpp:
* Source/JavaScriptCore/b3/B3CheckSpecial.h:
* Source/JavaScriptCore/b3/B3DataSection.cpp:
* Source/JavaScriptCore/b3/B3DataSection.h:
* Source/JavaScriptCore/b3/B3Dominators.h:
* Source/JavaScriptCore/b3/B3NaturalLoops.h:
* Source/JavaScriptCore/b3/B3PhiChildren.cpp:
* Source/JavaScriptCore/b3/B3PhiChildren.h:
* Source/JavaScriptCore/b3/B3Procedure.cpp:
* Source/JavaScriptCore/b3/B3Procedure.h:
* Source/JavaScriptCore/b3/B3StackmapSpecial.cpp:
* Source/JavaScriptCore/b3/B3StackmapSpecial.h:
* Source/JavaScriptCore/b3/B3TZoneImpls.cpp: Copied from Source/JavaScriptCore/b3/B3BackwardsDominators.h.
* Source/JavaScriptCore/b3/B3Value.cpp:
* Source/JavaScriptCore/b3/B3Value.h:
* Source/JavaScriptCore/b3/B3ValueRep.cpp:
* Source/JavaScriptCore/b3/B3ValueRep.h:
* Source/JavaScriptCore/b3/B3Variable.cpp:
* Source/JavaScriptCore/b3/B3Variable.h:
* Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackAndGenerateCode.cpp:
* Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackAndGenerateCode.h:
* Source/JavaScriptCore/b3/air/AirBasicBlock.cpp:
* Source/JavaScriptCore/b3/air/AirBasicBlock.h:
* Source/JavaScriptCore/b3/air/AirCCallSpecial.cpp:
* Source/JavaScriptCore/b3/air/AirCCallSpecial.h:
* Source/JavaScriptCore/b3/air/AirCFG.h:
* Source/JavaScriptCore/b3/air/AirCode.cpp:
* Source/JavaScriptCore/b3/air/AirCode.h:
* Source/JavaScriptCore/b3/air/AirDisassembler.cpp:
* Source/JavaScriptCore/b3/air/AirDisassembler.h:
* Source/JavaScriptCore/b3/air/AirLiveness.h:
* Source/JavaScriptCore/b3/air/AirLivenessAdapter.h:
* Source/JavaScriptCore/b3/air/AirPrintSpecial.cpp:
* Source/JavaScriptCore/b3/air/AirPrintSpecial.h:
* Source/JavaScriptCore/b3/air/AirSpecial.cpp:
* Source/JavaScriptCore/b3/air/AirSpecial.h:
* Source/JavaScriptCore/b3/air/AirStackSlot.cpp:
* Source/JavaScriptCore/b3/air/AirStackSlot.h:
* Source/JavaScriptCore/b3/air/AirTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/JavaScriptCore/builtins/BuiltinExecutables.cpp:
* Source/JavaScriptCore/builtins/BuiltinExecutables.h:
* Source/JavaScriptCore/builtins/BuiltinNames.cpp:
* Source/JavaScriptCore/builtins/BuiltinNames.h:
* Source/JavaScriptCore/bytecode/AccessCaseSnippetParams.cpp:
* Source/JavaScriptCore/bytecode/AccessCaseSnippetParams.h:
* Source/JavaScriptCore/bytecode/AdaptiveInferredPropertyValueWatchpointBase.cpp:
* Source/JavaScriptCore/bytecode/AdaptiveInferredPropertyValueWatchpointBase.h:
* Source/JavaScriptCore/bytecode/BytecodeBasicBlock.h:
* Source/JavaScriptCore/bytecode/BytecodeGraph.h:
* Source/JavaScriptCore/bytecode/BytecodeIntrinsicRegistry.cpp:
* Source/JavaScriptCore/bytecode/BytecodeIntrinsicRegistry.h:
* Source/JavaScriptCore/bytecode/BytecodeLivenessAnalysis.cpp:
* Source/JavaScriptCore/bytecode/BytecodeLivenessAnalysis.h:
* Source/JavaScriptCore/bytecode/CallLinkStatus.cpp:
* Source/JavaScriptCore/bytecode/CallLinkStatus.h:
* Source/JavaScriptCore/bytecode/CallVariant.h:
* Source/JavaScriptCore/bytecode/CheckPrivateBrandStatus.cpp:
* Source/JavaScriptCore/bytecode/CheckPrivateBrandStatus.h:
* Source/JavaScriptCore/bytecode/CheckPrivateBrandVariant.h:
* Source/JavaScriptCore/bytecode/CodeOrigin.cpp:
* Source/JavaScriptCore/bytecode/CodeOrigin.h:
* Source/JavaScriptCore/bytecode/DeferredSourceDump.h:
* Source/JavaScriptCore/bytecode/DeleteByStatus.cpp:
* Source/JavaScriptCore/bytecode/DeleteByStatus.h:
* Source/JavaScriptCore/bytecode/DeleteByVariant.h:
* Source/JavaScriptCore/bytecode/ExecutionCounter.cpp:
* Source/JavaScriptCore/bytecode/ExecutionCounter.h:
* Source/JavaScriptCore/bytecode/FullBytecodeLiveness.h:
* Source/JavaScriptCore/bytecode/GetByVariant.h:
* Source/JavaScriptCore/bytecode/InByStatus.cpp:
* Source/JavaScriptCore/bytecode/InByStatus.h:
* Source/JavaScriptCore/bytecode/InByVariant.h:
* Source/JavaScriptCore/bytecode/InstanceOfStatus.h:
* Source/JavaScriptCore/bytecode/InstanceOfVariant.h:
* Source/JavaScriptCore/bytecode/MetadataTable.h:
* Source/JavaScriptCore/bytecode/PutByVariant.h:
* Source/JavaScriptCore/bytecode/SetPrivateBrandStatus.cpp:
* Source/JavaScriptCore/bytecode/SetPrivateBrandStatus.h:
* Source/JavaScriptCore/bytecode/SetPrivateBrandVariant.h:
* Source/JavaScriptCore/bytecode/StructureStubClearingWatchpoint.cpp:
* Source/JavaScriptCore/bytecode/StructureStubClearingWatchpoint.h:
* Source/JavaScriptCore/bytecode/StructureStubInfo.h:
* Source/JavaScriptCore/bytecode/UnlinkedCodeBlockGenerator.cpp:
* Source/JavaScriptCore/bytecode/UnlinkedCodeBlockGenerator.h:
* Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp:
* Source/JavaScriptCore/bytecompiler/BytecodeGenerator.h:
* Source/JavaScriptCore/config.h:
* Source/JavaScriptCore/debugger/Breakpoint.h:
* Source/JavaScriptCore/debugger/Debugger.cpp:
* Source/JavaScriptCore/debugger/Debugger.h:
* Source/JavaScriptCore/dfg/DFGAbstractInterpreter.h:
* Source/JavaScriptCore/dfg/DFGAdaptiveInferredPropertyValueWatchpoint.cpp:
* Source/JavaScriptCore/dfg/DFGAdaptiveInferredPropertyValueWatchpoint.h:
* Source/JavaScriptCore/dfg/DFGArrayifySlowPathGenerator.h:
* Source/JavaScriptCore/dfg/DFGBackwardsCFG.h:
* Source/JavaScriptCore/dfg/DFGBackwardsDominators.h:
* Source/JavaScriptCore/dfg/DFGBasicBlock.cpp:
* Source/JavaScriptCore/dfg/DFGBasicBlock.h:
* Source/JavaScriptCore/dfg/DFGBlockMap.h:
* Source/JavaScriptCore/dfg/DFGCFG.h:
* Source/JavaScriptCore/dfg/DFGCSEPhase.cpp:
* Source/JavaScriptCore/dfg/DFGCallArrayAllocatorSlowPathGenerator.h:
* Source/JavaScriptCore/dfg/DFGCallCreateDirectArgumentsSlowPathGenerator.h:
* Source/JavaScriptCore/dfg/DFGCombinedLiveness.cpp:
* Source/JavaScriptCore/dfg/DFGCombinedLiveness.h:
* Source/JavaScriptCore/dfg/DFGControlEquivalenceAnalysis.h:
* Source/JavaScriptCore/dfg/DFGDisassembler.cpp:
* Source/JavaScriptCore/dfg/DFGDisassembler.h:
* Source/JavaScriptCore/dfg/DFGDominators.h:
* Source/JavaScriptCore/dfg/DFGFailedFinalizer.cpp:
* Source/JavaScriptCore/dfg/DFGFailedFinalizer.h:
* Source/JavaScriptCore/dfg/DFGFinalizer.cpp:
* Source/JavaScriptCore/dfg/DFGFinalizer.h:
* Source/JavaScriptCore/dfg/DFGFlowIndexing.cpp:
* Source/JavaScriptCore/dfg/DFGFlowIndexing.h:
* Source/JavaScriptCore/dfg/DFGFlowMap.h:
* Source/JavaScriptCore/dfg/DFGInPlaceAbstractState.cpp:
* Source/JavaScriptCore/dfg/DFGInPlaceAbstractState.h:
* Source/JavaScriptCore/dfg/DFGJITCompiler.cpp:
* Source/JavaScriptCore/dfg/DFGJITCompiler.h:
* Source/JavaScriptCore/dfg/DFGJITFinalizer.cpp:
* Source/JavaScriptCore/dfg/DFGJITFinalizer.h:
* Source/JavaScriptCore/dfg/DFGNaturalLoops.h:
* Source/JavaScriptCore/dfg/DFGNullAbstractState.h:
* Source/JavaScriptCore/dfg/DFGPhiChildren.cpp:
* Source/JavaScriptCore/dfg/DFGPhiChildren.h:
* Source/JavaScriptCore/dfg/DFGSaneStringGetByValSlowPathGenerator.h:
* Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp:
* Source/JavaScriptCore/dfg/DFGSpeculativeJIT.h:
* Source/JavaScriptCore/dfg/DFGTZoneImpls.cpp: Copied from Source/JavaScriptCore/dfg/DFGFlowIndexing.cpp.
* Source/JavaScriptCore/dfg/DFGVariableAccessData.h:
* Source/JavaScriptCore/disassembler/ARM64/A64DOpcode.cpp:
* Source/JavaScriptCore/disassembler/ARM64/A64DOpcode.h:
* Source/JavaScriptCore/disassembler/Disassembler.cpp:
* Source/JavaScriptCore/ftl/FTLAbstractHeap.cpp:
* Source/JavaScriptCore/ftl/FTLAbstractHeap.h:
* Source/JavaScriptCore/ftl/FTLJITFinalizer.cpp:
* Source/JavaScriptCore/ftl/FTLJITFinalizer.h:
* Source/JavaScriptCore/ftl/FTLLazySlowPath.cpp:
* Source/JavaScriptCore/ftl/FTLLazySlowPath.h:
* Source/JavaScriptCore/ftl/FTLThunks.cpp:
* Source/JavaScriptCore/ftl/FTLThunks.h:
* Source/JavaScriptCore/heap/AbstractSlotVisitor.h:
* Source/JavaScriptCore/heap/AlignedMemoryAllocator.cpp:
* Source/JavaScriptCore/heap/BlockDirectoryBits.h:
* Source/JavaScriptCore/heap/CodeBlockSet.cpp:
* Source/JavaScriptCore/heap/CodeBlockSet.h:
* Source/JavaScriptCore/heap/GCSegmentedArray.cpp:
* Source/JavaScriptCore/heap/GCSegmentedArray.h:
* Source/JavaScriptCore/heap/HeapCellType.h:
* Source/JavaScriptCore/heap/HeapProfiler.cpp:
* Source/JavaScriptCore/heap/HeapProfiler.h:
* Source/JavaScriptCore/heap/HeapSnapshot.cpp:
* Source/JavaScriptCore/heap/HeapSnapshot.h:
* Source/JavaScriptCore/heap/HeapSnapshotBuilder.cpp:
* Source/JavaScriptCore/heap/HeapSnapshotBuilder.h:
* Source/JavaScriptCore/heap/IsoHeapCellType.cpp:
* Source/JavaScriptCore/heap/IsoHeapCellType.h:
* Source/JavaScriptCore/heap/IsoSubspace.cpp:
* Source/JavaScriptCore/heap/IsoSubspace.h:
* Source/JavaScriptCore/heap/JITStubRoutineSet.cpp:
* Source/JavaScriptCore/heap/JITStubRoutineSet.h:
* Source/JavaScriptCore/heap/MachineStackMarker.cpp:
* Source/JavaScriptCore/heap/MachineStackMarker.h:
* Source/JavaScriptCore/heap/MarkStackMergingConstraint.cpp:
* Source/JavaScriptCore/heap/MarkStackMergingConstraint.h:
* Source/JavaScriptCore/heap/MarkingConstraint.cpp:
* Source/JavaScriptCore/heap/MarkingConstraint.h:
* Source/JavaScriptCore/heap/MarkingConstraintSet.cpp:
* Source/JavaScriptCore/heap/MarkingConstraintSet.h:
* Source/JavaScriptCore/heap/MarkingConstraintSolver.h:
* Source/JavaScriptCore/heap/MutatorScheduler.cpp:
* Source/JavaScriptCore/heap/MutatorScheduler.h:
* Source/JavaScriptCore/heap/SimpleMarkingConstraint.cpp:
* Source/JavaScriptCore/heap/SimpleMarkingConstraint.h:
* Source/JavaScriptCore/heap/SlotVisitor.cpp:
* Source/JavaScriptCore/heap/SlotVisitor.h:
* Source/JavaScriptCore/heap/SpaceTimeMutatorScheduler.cpp:
* Source/JavaScriptCore/heap/SpaceTimeMutatorScheduler.h:
* Source/JavaScriptCore/heap/StochasticSpaceTimeMutatorScheduler.cpp:
* Source/JavaScriptCore/heap/StochasticSpaceTimeMutatorScheduler.h:
* Source/JavaScriptCore/heap/Subspace.cpp:
* Source/JavaScriptCore/heap/Subspace.h:
* Source/JavaScriptCore/heap/SynchronousStopTheWorldMutatorScheduler.cpp:
* Source/JavaScriptCore/heap/SynchronousStopTheWorldMutatorScheduler.h:
* Source/JavaScriptCore/heap/VerifierSlotVisitor.cpp:
* Source/JavaScriptCore/heap/VerifierSlotVisitor.h:
* Source/JavaScriptCore/inspector/ConsoleMessage.cpp:
* Source/JavaScriptCore/inspector/ConsoleMessage.h:
* Source/JavaScriptCore/inspector/InjectedScriptManager.cpp:
* Source/JavaScriptCore/inspector/InjectedScriptManager.h:
* Source/JavaScriptCore/inspector/InspectorAgentBase.h:
* Source/JavaScriptCore/inspector/JSGlobalObjectConsoleClient.cpp:
* Source/JavaScriptCore/inspector/JSGlobalObjectConsoleClient.h:
* Source/JavaScriptCore/inspector/JSGlobalObjectDebugger.cpp:
* Source/JavaScriptCore/inspector/JSGlobalObjectDebugger.h:
* Source/JavaScriptCore/inspector/JSGlobalObjectInspectorController.cpp:
* Source/JavaScriptCore/inspector/JSGlobalObjectInspectorController.h:
* Source/JavaScriptCore/inspector/JSInjectedScriptHost.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorAuditAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorAuditAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorConsoleAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorConsoleAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorHeapAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorHeapAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorRuntimeAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorRuntimeAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorScriptProfilerAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorScriptProfilerAgent.h:
* Source/JavaScriptCore/inspector/agents/InspectorTargetAgent.cpp:
* Source/JavaScriptCore/inspector/agents/InspectorTargetAgent.h:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectAuditAgent.cpp:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectAuditAgent.h:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectDebuggerAgent.cpp:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectDebuggerAgent.h:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectRuntimeAgent.cpp:
* Source/JavaScriptCore/inspector/agents/JSGlobalObjectRuntimeAgent.h:
* Source/JavaScriptCore/interpreter/CheckpointOSRExitSideState.h:
* Source/JavaScriptCore/interpreter/Interpreter.h:
* Source/JavaScriptCore/interpreter/Register.h:
* Source/JavaScriptCore/interpreter/ShadowChicken.cpp:
* Source/JavaScriptCore/interpreter/ShadowChicken.h:
* Source/JavaScriptCore/jit/AssemblyHelpers.cpp:
* Source/JavaScriptCore/jit/AssemblyHelpers.h:
* Source/JavaScriptCore/jit/CCallHelpers.cpp:
* Source/JavaScriptCore/jit/CCallHelpers.h:
* Source/JavaScriptCore/jit/CallFrameShuffleData.cpp:
* Source/JavaScriptCore/jit/CallFrameShuffleData.h:
* Source/JavaScriptCore/jit/CallFrameShuffler.h:
* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
* Source/JavaScriptCore/jit/ExecutableAllocator.h:
* Source/JavaScriptCore/jit/ICStats.cpp:
* Source/JavaScriptCore/jit/ICStats.h:
* Source/JavaScriptCore/jit/JIT.cpp:
* Source/JavaScriptCore/jit/JIT.h:
* Source/JavaScriptCore/jit/JITCompilation.cpp:
* Source/JavaScriptCore/jit/JITCompilation.h:
* Source/JavaScriptCore/jit/JITDisassembler.cpp:
* Source/JavaScriptCore/jit/JITDisassembler.h:
* Source/JavaScriptCore/jit/JITMathIC.h:
* Source/JavaScriptCore/jit/JITOpaqueByproduct.h:
* Source/JavaScriptCore/jit/JITOpaqueByproducts.cpp:
* Source/JavaScriptCore/jit/JITOpaqueByproducts.h:
* Source/JavaScriptCore/jit/JITSizeStatistics.cpp:
* Source/JavaScriptCore/jit/JITSizeStatistics.h:
* Source/JavaScriptCore/jit/JITTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/JavaScriptCore/jit/JITThunks.cpp:
* Source/JavaScriptCore/jit/JITThunks.h:
* Source/JavaScriptCore/jit/JITWorklist.cpp:
* Source/JavaScriptCore/jit/JITWorklist.h:
* Source/JavaScriptCore/jit/JSInterfaceJIT.h:
* Source/JavaScriptCore/jit/PCToCodeOriginMap.cpp:
* Source/JavaScriptCore/jit/PCToCodeOriginMap.h:
* Source/JavaScriptCore/jit/SpecializedThunkJIT.h:
* Source/JavaScriptCore/jsc.cpp:
* Source/JavaScriptCore/llint/LLIntOffsetsExtractor.cpp:
* Source/JavaScriptCore/parser/Lexer.cpp:
* Source/JavaScriptCore/parser/Lexer.h:
* Source/JavaScriptCore/parser/ModuleScopeData.h:
* Source/JavaScriptCore/parser/Parser.cpp:
* Source/JavaScriptCore/parser/Parser.h:
* Source/JavaScriptCore/parser/VariableEnvironment.cpp:
* Source/JavaScriptCore/parser/VariableEnvironment.h:
* Source/JavaScriptCore/profiler/ProfilerDatabase.cpp:
* Source/JavaScriptCore/profiler/ProfilerDatabase.h:
* Source/JavaScriptCore/profiler/ProfilerExecutionCounter.h:
* Source/JavaScriptCore/profiler/ProfilerTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/JavaScriptCore/runtime/ArgList.cpp:
* Source/JavaScriptCore/runtime/ArgList.h:
* Source/JavaScriptCore/runtime/ArrayBuffer.h:
* Source/JavaScriptCore/runtime/BasicBlockLocation.cpp:
* Source/JavaScriptCore/runtime/BasicBlockLocation.h:
* Source/JavaScriptCore/runtime/BufferMemoryHandle.cpp:
* Source/JavaScriptCore/runtime/BufferMemoryHandle.h:
* Source/JavaScriptCore/runtime/CodeCache.cpp:
* Source/JavaScriptCore/runtime/CodeCache.h:
* Source/JavaScriptCore/runtime/CommonIdentifiers.cpp:
* Source/JavaScriptCore/runtime/CommonIdentifiers.h:
* Source/JavaScriptCore/runtime/ControlFlowProfiler.cpp:
* Source/JavaScriptCore/runtime/ControlFlowProfiler.h:
* Source/JavaScriptCore/runtime/DOMAnnotation.h:
* Source/JavaScriptCore/runtime/DeferredWorkTimer.cpp:
* Source/JavaScriptCore/runtime/DeferredWorkTimer.h:
* Source/JavaScriptCore/runtime/DoublePredictionFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/DoublePredictionFuzzerAgent.h:
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgent.h:
* Source/JavaScriptCore/runtime/FileBasedFuzzerAgentBase.h:
* Source/JavaScriptCore/runtime/GenericOffset.h:
* Source/JavaScriptCore/runtime/ISO8601.h:
* Source/JavaScriptCore/runtime/IntlCache.cpp:
* Source/JavaScriptCore/runtime/IntlCache.h:
* Source/JavaScriptCore/runtime/IntlNumberFormat.h:
* Source/JavaScriptCore/runtime/JSDateMath.cpp:
* Source/JavaScriptCore/runtime/JSDateMath.h:
* Source/JavaScriptCore/runtime/JSDestructibleObjectHeapCellType.cpp:
* Source/JavaScriptCore/runtime/JSDestructibleObjectHeapCellType.h:
* Source/JavaScriptCore/runtime/JSGlobalObject.h:
(JSC::JSGlobalObject::customGetterFunctionSet):
(JSC::JSGlobalObject::customSetterFunctionSet):
* Source/JavaScriptCore/runtime/JSGlobalObjectDebuggable.cpp:
* Source/JavaScriptCore/runtime/JSGlobalObjectDebuggable.h:
* Source/JavaScriptCore/runtime/JSRunLoopTimer.cpp:
* Source/JavaScriptCore/runtime/JSRunLoopTimer.h:
* Source/JavaScriptCore/runtime/MegamorphicCache.cpp:
* Source/JavaScriptCore/runtime/MegamorphicCache.h:
* Source/JavaScriptCore/runtime/NarrowingNumberPredictionFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/NarrowingNumberPredictionFuzzerAgent.h:
* Source/JavaScriptCore/runtime/NativeCallee.h:
* Source/JavaScriptCore/runtime/NativeCalleeRegistry.h:
* Source/JavaScriptCore/runtime/ObjectPropertyChangeAdaptiveWatchpoint.h:
* Source/JavaScriptCore/runtime/Options.cpp:
* Source/JavaScriptCore/runtime/PredictionFileCreatingFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/PredictionFileCreatingFuzzerAgent.h:
* Source/JavaScriptCore/runtime/PropertyDescriptor.cpp:
* Source/JavaScriptCore/runtime/RandomizingFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/RandomizingFuzzerAgent.h:
* Source/JavaScriptCore/runtime/RegExpCache.cpp:
* Source/JavaScriptCore/runtime/RegExpCache.h:
* Source/JavaScriptCore/runtime/RuntimeTZoneImpls.cpp: Copied from Source/JavaScriptCore/heap/IsoHeapCellType.cpp.
* Source/JavaScriptCore/runtime/SamplingProfiler.cpp:
* Source/JavaScriptCore/runtime/SamplingProfiler.h:
* Source/JavaScriptCore/runtime/SparseArrayValueMap.h:
* Source/JavaScriptCore/runtime/StringReplaceCache.h:
* Source/JavaScriptCore/runtime/StringSplitCache.h:
* Source/JavaScriptCore/runtime/StructureRareData.cpp:
* Source/JavaScriptCore/runtime/TypeProfiler.cpp:
* Source/JavaScriptCore/runtime/TypeProfiler.h:
* Source/JavaScriptCore/runtime/TypeProfilerLog.cpp:
* Source/JavaScriptCore/runtime/TypeProfilerLog.h:
* Source/JavaScriptCore/runtime/VM.h:
* Source/JavaScriptCore/runtime/WaiterListManager.cpp:
* Source/JavaScriptCore/runtime/WaiterListManager.h:
* Source/JavaScriptCore/runtime/Watchdog.cpp:
* Source/JavaScriptCore/runtime/Watchdog.h:
* Source/JavaScriptCore/runtime/WeakGCSet.h:
* Source/JavaScriptCore/runtime/WideningNumberPredictionFuzzerAgent.cpp:
* Source/JavaScriptCore/runtime/WideningNumberPredictionFuzzerAgent.h:
* Source/JavaScriptCore/runtime/WriteBarrier.h:
* Source/JavaScriptCore/tools/CellList.h:
* Source/JavaScriptCore/tools/CompilerTimingScope.cpp:
* Source/JavaScriptCore/tools/HeapVerifier.cpp:
* Source/JavaScriptCore/tools/HeapVerifier.h:
* Source/JavaScriptCore/tools/JSDollarVM.cpp:
* Source/JavaScriptCore/tools/VMInspector.cpp:
* Source/JavaScriptCore/tools/VMInspector.h:
* Source/JavaScriptCore/wasm/WasmB3IRGenerator.cpp:
* Source/JavaScriptCore/wasm/WasmBBQDisassembler.cpp:
* Source/JavaScriptCore/wasm/WasmBBQDisassembler.h:
* Source/JavaScriptCore/wasm/WasmCallee.cpp:
* Source/JavaScriptCore/wasm/WasmCallee.h:
* Source/JavaScriptCore/wasm/WasmFormat.h:
* Source/JavaScriptCore/wasm/WasmFunctionCodeBlockGenerator.cpp:
* Source/JavaScriptCore/wasm/WasmFunctionCodeBlockGenerator.h:
* Source/JavaScriptCore/wasm/WasmFunctionIPIntMetadataGenerator.cpp:
* Source/JavaScriptCore/wasm/WasmFunctionIPIntMetadataGenerator.h:
* Source/JavaScriptCore/wasm/WasmFunctionParser.h:
* Source/JavaScriptCore/wasm/WasmGlobal.cpp:
* Source/JavaScriptCore/wasm/WasmGlobal.h:
* Source/JavaScriptCore/wasm/WasmMemory.cpp:
(): Deleted.
* Source/JavaScriptCore/wasm/WasmMemory.h:
* Source/JavaScriptCore/wasm/WasmOSREntryData.h:
* Source/JavaScriptCore/wasm/WasmStreamingParser.h:
* Source/JavaScriptCore/wasm/WasmTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/JavaScriptCore/wasm/WasmTable.cpp:
* Source/JavaScriptCore/wasm/WasmTable.h:
* Source/JavaScriptCore/wasm/WasmTag.cpp:
* Source/JavaScriptCore/wasm/WasmTag.h:
* Source/JavaScriptCore/wasm/WasmThunks.cpp:
* Source/JavaScriptCore/wasm/WasmThunks.h:
* Source/JavaScriptCore/wasm/WasmTierUpCount.cpp:
* Source/JavaScriptCore/wasm/WasmTierUpCount.h:
* Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp:
(JSC::Wasm::TypeDefinition::tryCreateFunctionSignature):
(JSC::Wasm::TypeDefinition::tryCreateStructType):
(JSC::Wasm::TypeDefinition::tryCreateArrayType):
(JSC::Wasm::TypeDefinition::tryCreateRecursionGroup):
(JSC::Wasm::TypeDefinition::tryCreateProjection):
(JSC::Wasm::TypeDefinition::tryCreateSubtype):
* Source/JavaScriptCore/wasm/WasmTypeDefinition.h:
* Source/JavaScriptCore/wasm/WasmValueLocation.h:
* Source/JavaScriptCore/wasm/WasmWorklist.cpp:
* Source/JavaScriptCore/wasm/WasmWorklist.h:
* Source/JavaScriptCore/wasm/js/WebAssemblyFunction.cpp:
* Source/JavaScriptCore/yarr/RegularExpression.cpp:
* Source/JavaScriptCore/yarr/RegularExpression.h:
* Source/JavaScriptCore/yarr/YarrDisassembler.cpp:
* Source/JavaScriptCore/yarr/YarrDisassembler.h:
* Source/JavaScriptCore/yarr/YarrInterpreter.cpp:
* Source/JavaScriptCore/yarr/YarrInterpreter.h:
* Source/JavaScriptCore/yarr/YarrJIT.cpp:
* Source/JavaScriptCore/yarr/YarrJIT.h:
* Source/JavaScriptCore/yarr/YarrPattern.h:
* Source/JavaScriptCore/yarr/YarrTZoneImpls.cpp: Copied from Source/JavaScriptCore/wasm/WasmTag.cpp.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/CMakeLists.txt:
* Source/WTF/wtf/PlatformUse.h:
* Source/WTF/wtf/TZoneMalloc.h: Copied from Source/JavaScriptCore/jit/JITSizeStatistics.h.
* Source/WTF/wtf/TZoneMallocInlines.h: Copied from Source/JavaScriptCore/runtime/StringSplitCache.h.
* Source/bmalloc/bmalloc.xcodeproj/project.pbxproj:
* Source/bmalloc/bmalloc/TZoneHeap.cpp: Copied from Source/JavaScriptCore/runtime/StringReplaceCache.h.
(bmalloc::api::tzoneAllocate):
(bmalloc::api::tzoneTryAllocate):
(bmalloc::api::tzoneDeallocate):
* Source/bmalloc/bmalloc/TZoneHeap.h: Added.
(bmalloc::api::TZoneHeap::TZoneHeap):
(bmalloc::api::TZoneHeap::allocate):
(bmalloc::api::TZoneHeap::tryAllocate):
(bmalloc::api::TZoneHeap::deallocate):
(bmalloc::api::TZoneHeap::scavenge):
(bmalloc::api::TZoneHeap::initialize):
(bmalloc::api::TZoneHeap::isInitialized):
(bmalloc::api::TZoneHeap::provideHeap):
(bmalloc::api::TZoneHeap::allocatorOffset):
(bmalloc::api::TZoneHeap::setAllocatorOffset):
(bmalloc::api::TZoneHeap::deallocatorOffset):
(bmalloc::api::TZoneHeap::setDeallocatorOffset):
* Source/bmalloc/bmalloc/TZoneHeapInlines.h: Added.
(bmalloc::api::IsoHeap<Type>::TZoneHeap):
(bmalloc::api::TZoneHeap<Type>::allocate):
(bmalloc::api::TZoneHeap<Type>::tryAllocate):
(bmalloc::api::TZoneHeap<Type>::deallocate):
(bmalloc::api::TZoneHeap<Type>::scavenge):
(bmalloc::api::TZoneHeap<Type>::isInitialized):
(bmalloc::api::TZoneHeap<Type>::initialize):
(bmalloc::api::TZoneHeap<Type>::impl):

Canonical link: https://commits.webkit.org/271904@main
